apps/python/sampler/activities/lmeh/utils/mongodb.py

```diff
@@ -30,11 +43,60 @@ def reconstruct_instance(_id: str, collection: pymongo.collection.Collection):
     return instance
 
 
-def instance_to_dict(instance:Instance, task_id: ObjectId)-> dict:
-
+def instance_to_dict(instance: Instance, task_id: ObjectId)-> dict:
     instance_mongo = asdict(instance)
     instance_mongo.pop('resps', None)
     instance_mongo.pop('filtered_resps', None)
     instance_mongo['task_id'] = task_id
     instance_mongo['_id'] = ObjectId()
-    return instance_mongo            
\ No newline at end of file
+    instance_mongo['done'] = False
+    return instance_mongo
+
+def get_tokenizer_objects(
+        node_adress: str, client: pymongo.MongoClient, db_name:str='pocket-ml-testbench', 
+        nodes_collection_name:str='nodes', tokenizers_collection_name:str='tokenizers'
+        )-> dict:
+    
+    node = list(client[db_name][nodes_collection_name].find({'node': node_adress}))
+    eval_logger.debug(f"Node found.", node=node)
+    if len(node) == 0:
+        eval_logger.error(f"Node adress not found.", adress=node_adress)
+        raise ApplicationError(f"Node adress {node_adress} does not exist in the database.")    
+    elif len(node) > 1:
+        eval_logger.error(f"Multiple nodes found for adress.", adress=node_adress)
+        raise ApplicationError(f"Multiple nodes found for adress {node_adress}.")
```

Comment:
This should not be an error. If more than one service/chain is allowed per node and more than two ML services are pointing to the same node, for example a LLM and a MM, we will probably have two entries of the same node, one for the LLM service and one for the MM service.
Search for the node using Address and Service, you should have both from the task request.